apps/astarte_appengine_api/lib/astarte_appengine_api/device/device.ex

```diff
@@ -84,11 +84,27 @@ defmodule Astarte.AppEngine.API.Device do
         |> Map.put(:metadata, merged_metadata)
 
       {:ok, updated_map}
+    else
+      {:error, :invalid_alias} ->
+        {:error, :invalid_alias}
+
+      {:error, :invalid_metadata} ->
+        {:error, :invalid_metadata}
+
+      {:error, :alias_already_in_use} ->
+        {:error, :alias_already_in_use}
+
+      error ->
+        error
     end
```

Comment:
You're catching 3 special cases and returning them as-is, then you have a generic case that you also return as-is. Might as well drop the `else` completely since it will fall-through anyway